### ISSUE TITLE
add test coverage report

### ISSUE THREAD

**Issue by ltalirz** [aiidateam/aiida_core#1412]
Add report on test coverage, ideally automated for every new pull request.

**Comment by ltalirz** [comment]
@sphuber @muhrin I'm working on this in [my fork](https://github.com/ltalirz/aiida_core/commits/issue_1412_add_coverage_report) (branched off from `release_v0.12.0`) and I am getting a plumpy-related error in the sqlalchemy & django db tests:
```
2018-04-12 14:30:24,173 plum.persistence WARNING  Failed to load checkpoint '/tmp/tmpNfx_p5/running/206.pickle'
Traceback (most recent call last):
  File "/home/travis/build/ltalirz/aiida_core/aiida/work/persistence.py", line 177, in load_all_processes
    process = self.create_from_file_and_persist(f)
  File "/home/travis/build/ltalirz/aiida_core/aiida/work/persistence.py", line 222, in create_from_file_and_persist
    process = Process.create_from(checkpoint)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/plum/process.py", line 124, in create_from
    proc.perform_create(saved_instance_state=saved_instance_state)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/plum/process.py", line 366, in perform_create
    self.on_create(pid, inputs, saved_instance_state)
  File "/home/travis/build/ltalirz/aiida_core/aiida/backends/tests/work/daemon.py", line 106, in on_create
    raise RuntimeError()
RuntimeError
```
Any ideas what the reason for this could be?

I am running the tests in a slightly different way since `coverage run` expects a python file.
https://github.com/ltalirz/aiida_core/blob/ec23a6acd3fc99609c5b8e5c41d6d38d12079a46/.travis-data/test_script.sh#L25

https://github.com/ltalirz/aiida_core/blob/ec23a6acd3fc99609c5b8e5c41d6d38d12079a46/.travis-data/test_devel.py#L1-L10

The full travis logs are here: https://travis-ci.org/ltalirz/aiida_core/jobs/365637866#L796

**Comment by sphuber** [comment]
It is a warning and doesn't actually fail the tests right? It has been there for a long time and is part of the tests and expected I think

**Comment by ltalirz** [comment]
@sphuber You're right, thanks for the pointer. Problem found!

**Comment by sphuber** [comment]
Fixed in #1422
